third_party/move/move-vm/runtime/src/storage/verifier.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::loader::Module;
+use move_binary_format::{errors::PartialVMResult, file_format::CompiledScript, CompiledModule};
+
+/// Represents a verifier which is used for loaded modules and scripts. Clients
+/// can implement their own verification logic in addition to the bytecode verifier,
+/// and ensure that their passes are also running after.
```

Comment:
is there some clean place where verifier passes should be registered, if so might be useful to mention where.

Reply:
The way I see it is that the client implements this, including scheduling the bytecode verifier.

Reply:
oh good point, it's on the client side, makes sense

Reply:
i'm used to code in the aptos-vm lol

Reply:
It is less ugly there as well :))